source/assets/components/RecipeCard.js

```diff
@@ -0,0 +1,273 @@
+  const input = document.createElement("input");
+    let liked = true;
+    input.classList.add("favorite");
+    input.type = "image";
+    input.setAttribute("src", "assets/images/heart.png");
+    
+    input.addEventListener("click", changeHeart);
+    
+    /**
+     * Toggle on and off the favorite heart button
+     * @returns void
+     */
+    function changeHeart()
+    {
+      if (liked)
+      {
+          input.setAttribute("src", "assets/images/empty_heart.png");
+          liked = false;
+      }
+      else
+      {
+          input.setAttribute("src", "assets/images/heart.png");
+          liked = true;
+      }
+    }
+
+    card.appendChild(input);
+
+  // p element - title
+  const recipeTitle = document.createElement("p");
+  recipeTitle.classList.add("title");
+  recipeTitle.textContent = searchForKey(data, "headline");
+  card.appendChild(recipeTitle);
+
+  //time element
+  const time = document.createElement("time");
+  const totalTime = searchForKey(data, "totalTime");
+  time.textContent = convertTime(totalTime);
+  card.appendChild(time);
+
+  // button element - check recipe
+  const checkButton = document.createElement("button");
+  checkButton.classList.add("cook");
+  checkButton.textContent = "View Recipe  ";
+  card.appendChild(checkButton);
+  checkButton.addEventListener("click", function (event) {
+    showCheck(data);
+  });
+
+   this.shadowRoot.appendChild(styleElem);
+   this.shadowRoot.appendChild(card);
+
+  }
+}
+
+
+/**
+ * Recursively search for a key nested somewhere inside an object
+ * @param {Object} object the object with which you'd like to search
+ * @param {String} key the key that you are looking for in the object
+ * @returns {*} the value of the found key
+ */
+function searchForKey(object, key) {
+  var value;
+  Object.keys(object).some(function (k) {
+    if (k === key) {
+      value = object[k];
+      return true;
+    }
+    if (object[k] && typeof object[k] === 'object') {
+      value = searchForKey(object[k], key);
+      return value !== undefined;
+    }
+  });
+  return value;
+}
+
+/**
+ * Extract the URL from the given recipe schema JSON object
+ * @param {Object} data Raw recipe JSON to find the URL of
+ * @returns {String} If found, it returns the URL as a string, otherwise null
+ */
+function getUrl(data) {
+  if (data.url) return data.url;
+  if (data['@graph']) {
+    for (let i = 0; i < data['@graph'].length; i++) {
+      if (data['@graph'][i]['@type'] == 'Article') return data['@graph'][i]['@id'];
+    }
+  };
+  return null;
+}
+
+/**
+ * Similar to getUrl(), this function extracts the organizations name from the
+ * schema JSON object. It's not in a standard location so this function helps.
+ * @param {Object} data Raw recipe JSON to find the org string of
+ * @returns {String} If found, it retuns the name of the org as a string, otherwise null
+ */
+function getOrganization(data) {
+  if (data.publisher?.name) return data.publisher?.name;
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/cse110-fa21-group16/cse110-fa21-group16/pullRequest?prid=8394473)
```suggestion
  if (data.publisher?.name) {return data.publisher?.name;}
```